### ISSUE TITLE
Question about loss function

### ISSUE THREAD

**Issue by cljosegfer** [pluckpeng/Audio2Face#1]
Hi, thank you for sharing your code.
Could you explain why you are using L1 as the loss function rather than MSE?
I've find it particular interesting because I assume your model's output is the blendshape, instead of the 3D position vertex from the original A2F paper.
Is there any concept idea behind this design choice?
Many thanks again.

**Comment by pluckpeng** [comment]
Hello, josegfer. Thanks for your issue.

### Why L1 loss?
If you have read the paper you might find that the loss function they design is a little bit complex and the whole function is largely depending on MSE. Actually what I want to say is there is no much difference between L1(MAE) and L2(MSE). But they have some difference when using different dataset. The dataset I used could have some errors and noises  hard to cope with. So I recommend you to read this paper for the details [Loss Functions for Image Restoration with Neural Networks](https://arxiv.org/pdf/1511.08861.pdf). Experiments show that in such a dataset it's better to use L1 loss for better tolerance to those errors.

### Why blendshape?
As for the second question, I have to admit that the reason I choose blendshape as output is because this is the only dataset I got from a competition. For other aspects, I guess it's much more easier to collect blendshape data than the 3D position vertex. I've try this with IPhone. Seems that all work could be done with a deep camera. Combined with Maya, Unity3D or Unreal, it's easy to make some digital humans with blenshape. Recently I heard that Meta is gonna make their dataset open source which might offer the 3D data you mentioned. You can check for more details.

BTW, I delete the emotional part from the original paper is just because it had bad results on this dataset. And if you're interested in A2F, I recommend you to check this repo [FACEGOOD](https://github.com/FACEGOOD/FACEGOOD-Audio2Face). Have a nice day!
